File: Day - 02/rotate matrix 90 deg.cpp
// /*
//     Rotate the matrix 90 deg
//     Input: 1 2 3
//            4 5 6
//            7 8 9
//     Output: 7 4 1
//             8 5 2
//             9 6 3
// */

#include<iostream>
#include<algorithm>
using namespace std;

int main()
{
    int arr[3][3] = {{1,2,3},{4,5,6},{7,8,9}};

     // transpose matrix
    for(int i=0; i<3; i++)
    {
        for(int j=0; j<i; j++)
        {
            int temp = arr[i][j];
            arr[i][j] = arr[j][i];
            arr[j][i] = temp;
        }
    }

    // reverse each row

    for(int i=0; i<3; i++)
    {
        reverse(arr[i],arr[i]+3);
    }

    // print the output of the matrix
    for(int i=0;i<3;i++)
    {
        for(int j=0;j<3;j++)
        {
            cout<<arr[i][j]<<" ";
        }
        cout<<endl;
    }
}

File: Day - 02/rotate matrix 90 deg.py
# Rotate the matrix 90 degrees
#     Input: 1 2 3
#            4 5 6
#            7 8 9
#     Output: 7 4 1
#             8 5 2
#             9 6 3

l = [[1,2,3],[4,5,6],[7,8,9]]

for i in range(3):
    for j in range(i):
        l[i][j],l[j][i] = l[j][i],l[i][j]

for i2 in range(3):
    l[i2].reverse()

for i1 in range(3):
    for j1 in range(3):
        print(l[i1][j1],end=" ")
    print("\n")